addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,498 @@
+                alt_id_type = '04'
+            else:
+                alt_id_type = '06'
+            if partner.country_id:
+                alt_id_country = partner.country_id.code
+
+        values_dest = {
+            'nif': nif,
+            'alt_id_country': alt_id_country,
+            'alt_id_number': alt_id_number,
+            'alt_id_type': alt_id_type,
+            'partner': partner,
+            'partner_address': ', '.join(filter(lambda x: x, [partner.street, partner.street2, partner.city])),
+        }
+
+        values.update({
+            'recipient': values_dest,
+            # TODO for Bizkaia, option below can be "T" (if "in" invoice)
+            'thirdparty_or_recipient': "D",  # thirdparty = Tercero, recipient = Destinatario
+        })
+        return values
+
+    def _l10n_es_tbai_get_invoice_values(self, invoice, cancel):
+        eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
+
+        # === HEADER (CABECERA) ===
+        values = {'invoice': invoice}
+        if cancel:
+            return values
+
+        # === CREDIT NOTE (RECTIFICATIVA) ===
+        # TODO values below would have to be adapted for in_invoices (Bizkaia LROE)
+        values['is_refund'] = invoice.move_type == 'out_refund'
+        values['is_simplified'] = invoice.partner_id == self.env.ref("l10n_es_edi_sii.partner_simplified")
+        if values['is_refund']:
+            # TODO check refund codes are legal before sending ? Do not use "defaults" here, show them in wizard
+            values['credit_note_code'] = invoice.l10n_es_tbai_refund_reason or ('R5' if values['is_simplified'] else 'R1')
+            values['credit_note_invoice'] = invoice.reversed_entry_id
+
+        # === LINES (DETALLES) ===
+        values['invoice_lines'] = [{
+            'line': line,
+            'address': regex_sub(r'[^0-9a-zA-Z ]', '', line.name)[:250],
+        } for line in invoice.invoice_line_ids.filtered(lambda line: not line.display_type)]
+
+        # CODES (CLAVES): TODO there's 15 more codes to implement, also there can be up to 3 in total
+        # See https://www.gipuzkoa.eus/documents/2456431/13761128/Anexo+I.pdf/2ab0116c-25b4-f16a-440e-c299952d683d
+        com_partner = invoice.commercial_partner_id
+        if not com_partner.country_id or com_partner.country_id.code in eu_country_codes:
+            values['vat_regime_code'] = '01'
+        else:
+            values['vat_regime_code'] = '02'
+
+        # === BREAKDOWN TYPE (TIPO DESGLOSE) ===
+        edi_format_sii = self.env['account.edi.format']
+        if com_partner.country_id.code in ('ES', False) and not (com_partner.vat or '').startswith("ESN"):
+            tax_details_info_vals = edi_format_sii._l10n_es_edi_get_invoices_tax_details_info(invoice)
+            values['invoice_breakdown'] = tax_details_info_vals['tax_details_info']
+            values['invoice_total'] = round(-1 * (tax_details_info_vals['tax_details']['base_amount']
```

Comment:
Not sure about "-1": This is correct only if you need to report the credit note amounts with a negative sign

Reply:
To be tested